src/Couchbase.Lite.Shared/API/Database/Collection.cs

```diff
@@ -269,13 +321,109 @@ public void DeleteIndex([NotNull] string name)
 
         #endregion
 
-        #region IDispose
+        #region Public Methods - QueryFactory
 
-        public void Dispose()
+        public IQuery CreateQuery([NotNull] string queryExpression)
         {
             throw new NotImplementedException();
         }
 
         #endregion
+
+        #region Internal Methods
+
+        internal void GetSpec()
+        {
+            ThreadSafety.DoLocked(() =>
+            {
+                var spec = Native.c4coll_getSpec(c4coll);
+                Name = spec.name.CreateString();
+                Scope.Name = spec.scope.CreateString();
+            });
+        }
+
+        /// <summary>
+        /// Returns false if this collection has been deleted, or its database closed.
+        /// </summary>
+        internal void CheckCollectionValid()
+        {
+            ThreadSafety.DoLocked(() =>
+            {
+                if (c4Db == null) {
+                    throw new InvalidOperationException(CouchbaseLiteErrorMessage.DBClosed);
+                }
+
+                var isValid = Native.c4coll_isValid((C4Collection*)_c4coll);
+                if (!isValid || _c4coll == IntPtr.Zero) {
```

Comment:
I will switch the check order here if this is what you mean?

Reply:
Calling c4coll_isValid on nullptr is probably going to crash, that's the problem.